D3D11 texture serialisation size estimation fix.

vk2gpu opened the pull request:
During size estimation phase of serialisation, the row + depth pitch were not conservative enough estimates on some drivers. To remedy this, it's determined exactly at estimation time by attempting to map the resources.

clang-format ran, hopefully the commit message is correct now.

Commit:
D3D11 texture serialisation size estimation fix.

During size estimation phase of serialisation, the row + depth pitch were not conservative enough estimates on some drivers. To remedy this, it's determined exactly at estimation time by attempting to map the resources.

baldurk commented on renderdoc/driver/d3d11/d3d11_initstate.cpp:
```diff
@@ -377,20 +386,27 @@ uint32_t WrappedID3D11Device::GetSize_InitialState(ResourceId id, ID3D11DeviceCh
 
       const UINT DepthPitch = WorstRowPitch * numRows;
-      const UINT WorstDepthPitch = AlignUp(DepthPitch, WorstDepthPitchAlign);
+      const UINT WorstDepthPitch = AlignUp(DepthPitch, pitch.m_DepthPitch);
```
Looking at this again now, I'm not sure I understand the logic here.

If we have the actual row/depth pitch they should just be used directly, not applied as an alignment on the calculated tightly-packed row width. In many cases the row or depth pitch from the driver won't be a power of two so the align calculation won't give the result we want.

The only reason I had 'aligning' here before was because I needed some way to estimate how much padding there would be on top of the tightly packed calculation of how many bytes a row contained at minimum. Since I was mostly concerned about alignment of the rows or padding for small mips, I added a pessimistic alignment to apply.

vk2gpu replied:
Yep, you're right. That is stupid naive code. Will patch that up. Comes from me just going "replace the old value with the new one" whilst running on some tiredness.

Commit:
Remove manual pitch calculation.

Since the pitch is calculated by GetResourcePitchForSubresource, there is no need to calculate it manually, or even erroneously as were the case by using the resource pitch determined by the D3D11 API.
